src/plugins/syscalls/scproto.h

```diff
@@ -105,414 +105,2979 @@
 /*
  * From http://laredo-13.mit.edu/~brendan/scproto.txt
  *
- * TODO Integrate argument extraction into syscall monitoring
  */
 
-#ifndef SYSCALLS_PROTOTYPES_H
-#define SYSCALLS_PROTOTYPES_H
+typedef struct
+{
+  const char* name;
+  const char* dir;
+  const char* dir_opt;
+  const char* type;
```

Comment:
Type should also not be a char*, an distinct type from an enum would make it faster to build on top (not having to do a bunch of slow `strcmp` down the line).